web/src/games/cardtable/game.ts

```diff
@@ -0,0 +1,194 @@
+import { Ctx } from 'boardgame.io';
+import { INVALID_MOVE } from 'boardgame.io/core';
+import { dealCribbage } from './deals';
+
+export interface IG {
+  count: number;
+  deck: ICard[];
+  hands: ITable;
+  stock: ICard[];
+}
+
+export interface ITable {
+  north: IHand;
+  south: IHand;
+  east: IHand;
+  west: IHand;
+}
+
+export interface ICard {
+  id: string;
+  rank: number;
+  faced: boolean;
+  img: string;
+}
+
+export interface IHand {
+  private: ICard[];
+  held: ICard[];
+  melds: ICard[];
+  tricks: ICard[];
+  played: ICard[];
+  cribFlipped?: boolean;
+}
+
+export interface ILocation {
+  container: string;
```

Comment:
instead of using a string and spliting each part with a dot, can you use an array instead?

Reply:
again this is a first attempt before understanding the total of constraints the design will require.  I'm not opposed to naming conventions that are checked in automation (linting) and well expressed in the docs.  I needed a way to traverse the object graph that was readable and easy to understand.  I'm not sure splitting the path into multiple keys achieves that, but willing to look at suggestions. Again this is a primitive card abstraction and is likely to move to common (or be replaced by a better version).

Reply:
This isn't about naming conventions, it is about using the language correctly. Strings are meant to store text, not data structures.

I won't block your game on this, but using dividers and splitting text to store data structures is a really bad practice.